opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -353,8 +353,11 @@ namespace Opm {
             // do the initialization for all the wells
             // TODO: to see whether we can postpone of the intialization of the well containers to
             // optimize the usage of the following several member variables
+            std::vector< Scalar > B_avg(numComponents(), Scalar() );
+            computeAverageFormationFactor(B_avg);
+
             for (auto& well : well_container_) {
-                well->init(&phase_usage_, depth_, gravity_, local_num_cells_);
+                well->init(&phase_usage_, depth_, gravity_, local_num_cells_, B_avg);
             }
```

Comment:
@GitPaean @atgeirr What do you think about this? Computing the B_avg at every well iteration and everytime we compute the well potential is really not necessary IMO.  I would happily remove the passing of the B_avg around in the rest of the well code if this gets merged. We shouldn't compute global averages like this more often than we need.